bin/improver-spot-extract-new

```diff
@@ -0,0 +1,130 @@
+        description="Extract diagnostic data from gridded fields for spot data"
+        " sites. It is possible to apply a temperature lapse rate adjustment"
+        " to temperature data that helps to account for differences between"
+        " the spot sites real altitude and that of the grid point from which"
+        " the temperature data is extracted.")
+
+    # Input and output files required.
+    parser.add_argument("neighbour_filepath", metavar="NEIGHBOUR_FILEPATH",
+                        help="Path to a NetCDF file of spot-data neighbours. "
+                        "This file also contains the spot site information.")
+    parser.add_argument("diagnostic_filepath", metavar="DIAGNOSTIC_FILEPATH",
+                        help="Path to a NetCDF file containing the diagnostic "
+                             "data to be extracted.")
+    parser.add_argument("output_filepath", metavar="OUTPUT_FILEPATH",
+                        help="The output path for the resulting NetCDF")
+
+    method_group = parser.add_argument_group("Neighbour finding method")
+    method_group.add_argument(
+        "--land_constraint", default=False, action='store_true',
+        help="If set the neighbour cube will be interogated for grid point"
+        " neighbours that were identified using a land constraint. This means"
+        " that the grid points should be land points except for sites where"
+        " none was found within the search radius when the neighbour cube was"
+        " created. May be used with minimum_dz.")
+    method_group.add_argument(
+        "--minimum_dz", default=False, action='store_true',
+        help="If set the neighbour cube will be interogated for grid point"
+        " neighbours that were identified using a minimum height difference"
+        " constraint. These are grid points that were found to be the closest"
+        " in altitude to the spot site within the search radius defined when"
+        " the neighbour cube was created. May be used with land_constraint.")
+
+    lapse_group = parser.add_argument_group(
+        "Temperature lapse rate adjustment")
+    lapse_group.add_argument(
+        "--temperature_lapse_rate_filepath",
+        help="Filepath to a NetCDF file containing temperature lapse rates. "
+        "If this cube is provided, and a screen temperature cube is being "
+        "processed, the lapse rates will be used to adjust the temperatures "
+        "to better represent each spot site's altitude.")
+
+    meta_group = parser.add_argument_group("Metadata")
+    meta_group.add_argument(
+        "--grid_metadata_identifier", default="mosg",
+        help="A string to identify attributes from the input netCDF files that"
+        " should be compared to ensure that the data is compatible. Spot"
+        " data works using grid indices, so it is important that the grids"
+        " are matching or the data extracted may not match the location of the"
+        " spot data sites.")
+
```

Comment:
One of the acceptance criteria for this ticket is to be able to adjust the metadata on the output spot cube.  Could you please add an optional json file argument to the "Metadata" group (similar to the "improver-standardise" CLI) to allow this to be manipulated in the future?

Reply:
I'm doing this and will add a test. There doesn't seem to be any good documentation for this apart from in the function doc-string. We should endeavour to add some Sphinx documentation for this functionality. I will write a ticket to this effect unless you can point me to something that already exists?

Reply:
I don't know of any existing functionality, no.  Gavin would be the person to ask.